### PR TITLE
Update Service Bus SDK to T2

### PR DESCRIPTION
**Pull request by v-yilinhu**
1. Update Service Bus SDK from t1 to t2

2. Update files:

     - ./samples/ingestion/ingestion-client/Connector/Connector.csproj
     - ./samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs
     - ./samples/ingestion/ingestion-client/FetchTranscription/FetchTranscription.csproj
     - ./samples/ingestion/ingestion-client/FetchTranscription/FetchTranscriptionEnvironmentVariables.cs
     - ./samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs
     - ./samples/ingestion/ingestion-client/StartTranscriptionByServiceBus/StartTranscriptionByServiceBus.cs
     - ./samples/ingestion/ingestion-client/StartTranscriptionByServiceBus/StartTranscriptionByServiceBus.csproj
     - ./samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.cs
     - ./samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.csproj
     - ./samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionEnvironmentVariables.cs
     - ./samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs

@scottaddie, @AlexGhiondea, @lilyjma, @ramya-rao-a and @JoshLove-msft for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @lilyjma, @ramya-rao-a and @JoshLove-msft Could you please review this PR?

**Comment by ramya-rao-a** [comment]
@JoshLove-msft Please review

File: samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs [hunk 1]
```diff
@@ -8,26 +8,27 @@ namespace Connector
...
     using Microsoft.Extensions.Logging;
 
     public static class ServiceBusUtilities
     {
-        public static async Task SendServiceBusMessageAsync(QueueClient queueClient, string messageContent, ILogger log, TimeSpan delay)
+        public static async Task SendServiceBusMessageAsync(ServiceBusSender queueSender, string messageContent, ILogger log, TimeSpan delay)
         {
-            var message = new Message(Encoding.UTF8.GetBytes(messageContent));
-            await SendServiceBusMessageAsync(queueClient, message, log, delay).ConfigureAwait(false);
+            var message = new Azure.Messaging.ServiceBus.ServiceBusMessage(Encoding.UTF8.GetBytes(messageContent));
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs:19]:
```suggestion
            var message = new ServiceBusMessage(messageContent);
```

File: samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs [hunk 2]
```diff
@@ -8,26 +8,27 @@ namespace Connector
...
+            await SendServiceBusMessageAsync(queueSender, message, log, delay).ConfigureAwait(false);
         }
 
-        public static async Task SendServiceBusMessageAsync(QueueClient queueClient, Message message, ILogger log, TimeSpan delay)
+        public static async Task SendServiceBusMessageAsync(ServiceBusSender queueSender, Azure.Messaging.ServiceBus.ServiceBusMessage message, ILogger log, TimeSpan delay)
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs:23]:
```suggestion
        public static async Task SendServiceBusMessageAsync(ServiceBusSender queueSender, ServiceBusMessage message, ILogger log, TimeSpan delay)
```

File: samples/ingestion/ingestion-client/FetchTranscription/FetchTranscriptionEnvironmentVariables.cs [hunk 3]
```diff
@@ -40,12 +40,16 @@ public static class FetchTranscriptionEnvironmentVariables
...
 
+        public static readonly string FetchTranscriptionServiceBusQueueName = Environment.GetEnvironmentVariable(nameof(FetchTranscriptionServiceBusQueueName), EnvironmentVariableTarget.Process);
+
         public static readonly string HtmlResultOutputContainer = Environment.GetEnvironmentVariable(nameof(HtmlResultOutputContainer), EnvironmentVariableTarget.Process);
 
         public static readonly string JsonResultOutputContainer = Environment.GetEnvironmentVariable(nameof(JsonResultOutputContainer), EnvironmentVariableTarget.Process);
 
         public static readonly string StartTranscriptionServiceBusConnectionString = Environment.GetEnvironmentVariable(nameof(StartTranscriptionServiceBusConnectionString), EnvironmentVariableTarget.Process);
 
+        public static readonly string StartTranscriptionServiceBusQueueName = Environment.GetEnvironmentVariable(nameof(StartTranscriptionServiceBusQueueName), EnvironmentVariableTarget.Process);
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/FetchTranscription/FetchTranscriptionEnvironmentVariables.cs:51]: Why did these need to be added?

Comment (JoshLove-msft) [reply samples/ingestion/ingestion-client/FetchTranscription/FetchTranscriptionEnvironmentVariables.cs:51]: Ah, I guess the names were previously picked up from the connection strings. We should be able to just use the ConnectionStringProperties type to parse out the queue name.
https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusconnectionstringproperties.entitypath?view=azure-dotnet#Azure_Messaging_ServiceBus_ServiceBusConnectionStringProperties_EntityPath

File: samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs [hunk 4]
```diff
@@ -145,7 +148,7 @@ private static async Task ProcessFailedTranscriptionAsync(string transcriptionLo
                 if (retryAudioFile && audio.RetryCount < FetchTranscriptionEnvironmentVariables.RetryLimit)
                 {
                     log.LogInformation($"Retrying transcription with name {fileName} - retry count: {audio.RetryCount}");
-                    var sbMessage = new ServiceBusMessage
+                    var sbMessage = new Connector.ServiceBusMessage
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs:151]: Why is this being added?

File: samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs [hunk 5]
```diff
@@ -155,8 +158,8 @@ private static async Task ProcessFailedTranscriptionAsync(string transcriptionLo
                         RetryCount = audio.RetryCount + 1
                     };
 
-                    var audioFileMessage = new Message(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(sbMessage)));
-                    await ServiceBusUtilities.SendServiceBusMessageAsync(StartQueueClientInstance, audioFileMessage, log, TimeSpan.FromMinutes(1)).ConfigureAwait(false);
+                    var audioFileMessage = new Azure.Messaging.ServiceBus.ServiceBusMessage(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(sbMessage)));
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs:161]:
```suggestion
                    var audioFileMessage = new ServiceBusMessage(JsonConvert.SerializeObject(sbMessage));
```

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.cs [hunk 6]
```diff
@@ -51,21 +54,21 @@ public static async Task Run([TimerTrigger("0 */2 * * * *")] TimerInfo myTimer,
...
                             validServiceBusMessages.Add(message);
                         }
                         else
                         {
-                            await MessageReceiverInstance.CompleteAsync(message.SystemProperties.LockToken).ConfigureAwait(false);
+                            await ServiceBusReceivedMessage.CompleteMessageAsync(message).ConfigureAwait(false);
                         }
                     }
-                    catch (MessageLockLostException)
+                    catch (ServiceBusException) when (nameof(ServiceBusException.Reason).Equals(ServiceBusFailureReason.MessageLockLost))
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.cs:71]:
```suggestion
                    catch (ServiceBusException ex) 
                    when (ex.Reason == ServiceBusFailureReason.MessageLockLost)
```

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.cs [hunk 7]
```diff
@@ -9,16 +9,19 @@ namespace StartTranscriptionByTimer
...
     public static class StartTranscriptionByTimer
     {
         private const double MessageReceiveTimeoutInSeconds = 60;
 
-        private static readonly MessageReceiver MessageReceiverInstance = new (new ServiceBusConnectionStringBuilder(StartTranscriptionEnvironmentVariables.StartTranscriptionServiceBusConnectionString), prefetchCount: StartTranscriptionEnvironmentVariables.MessagesPerFunctionExecution);
+        private static readonly ServiceBusClient ServiceBusClient = new ServiceBusClient(StartTranscriptionEnvironmentVariables.StartTranscriptionServiceBusConnectionString);
+
+        private static readonly ServiceBusReceiverOptions ServiceBusReceiverOptions = new ServiceBusReceiverOptions() { PrefetchCount = StartTranscriptionEnvironmentVariables.MessagesPerFunctionExecution };
+
+        private static readonly ServiceBusReceiver ServiceBusReceivedMessage = ServiceBusClient.CreateReceiver(StartTranscriptionEnvironmentVariables.StartTranscriptionServiceBusQueueName, ServiceBusReceiverOptions);
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionByTimer.cs:24]:
```suggestion
        private static readonly ServiceBusReceiver ServiceBusReceiver = ServiceBusClient.CreateReceiver(StartTranscriptionEnvironmentVariables.StartTranscriptionServiceBusQueueName, ServiceBusReceiverOptions);
```

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionEnvironmentVariables.cs [hunk 8]
```diff
@@ -48,12 +48,16 @@ public static class StartTranscriptionEnvironmentVariables
 
         public static readonly string FetchTranscriptionServiceBusConnectionString = Environment.GetEnvironmentVariable(nameof(FetchTranscriptionServiceBusConnectionString), EnvironmentVariableTarget.Process);
 
+        public static readonly string FetchTranscriptionServiceBusQueueName = Environment.GetEnvironmentVariable(nameof(FetchTranscriptionServiceBusQueueName), EnvironmentVariableTarget.Process);
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionEnvironmentVariables.cs:51]: These can be removed in favor of using ConnectionStringProperties parsing.

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs [hunk 9]
```diff
@@ -164,7 +167,7 @@ private async Task StartBatchTranscriptionJobAsync(IEnumerable<Message> messages
             }
 
             var locationString = string.Empty;
-            var serviceBusMessages = messages.Select(message => JsonConvert.DeserializeObject<ServiceBusMessage>(Encoding.UTF8.GetString(message.Body)));
+            var serviceBusMessages = messages.Select(message => JsonConvert.DeserializeObject<Connector.ServiceBusMessage>(Encoding.UTF8.GetString(message.Body)));
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs:170]: Why needed?

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs [hunk 10]
```diff
@@ -244,19 +247,19 @@ private async Task StartBatchTranscriptionJobAsync(IEnumerable<Message> messages
...
-                var sbMessage = JsonConvert.DeserializeObject<ServiceBusMessage>(Encoding.UTF8.GetString(message.Body));
+                var sbMessage = JsonConvert.DeserializeObject<Connector.ServiceBusMessage>(Encoding.UTF8.GetString(message.Body));
 
                 if (sbMessage.RetryCount <= StartTranscriptionEnvironmentVariables.RetryLimit || statusCode == HttpStatusCode.TooManyRequests)
                 {
                     sbMessage.RetryCount += 1;
                     var messageDelay = GetMessageDelayTime(sbMessage.RetryCount);
-                    var newMessage = new Message(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(sbMessage)));
-                    await ServiceBusUtilities.SendServiceBusMessageAsync(StartQueueClientInstance, newMessage, Logger, messageDelay).ConfigureAwait(false);
+                    var newMessage = new Azure.Messaging.ServiceBus.ServiceBusMessage(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(sbMessage)));
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs:261]:
```suggestion
                    var newMessage = new ServiceBusMessage(JsonConvert.SerializeObject(sbMessage));
```

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs [hunk 11]
```diff
@@ -269,7 +272,7 @@ private async Task RetryOrFailMessagesAsync(IEnumerable<Message> messages, strin
             }
         }
 
-        private async Task WriteFailedJobLogToStorageAsync(IEnumerable<ServiceBusMessage> serviceBusMessages, string errorMessage, string jobName)
+        private async Task WriteFailedJobLogToStorageAsync(IEnumerable<Connector.ServiceBusMessage> serviceBusMessages, string errorMessage, string jobName)
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs:275]: Why needed?

File: samples/ingestion/ingestion-client/Connector/Connector.csproj [hunk 12]
```diff
@@ -18,6 +18,7 @@
   </PropertyGroup>
   
   <ItemGroup>
+    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.5.0" />
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/Connector/Connector.csproj:21]: The Microsoft.Azure.WebJobs.Extensions.ServiceBus dependency needs to be updated to 5.1.0.

File: samples/ingestion/ingestion-client/FetchTranscription/FetchTranscription.csproj [hunk 13]
```diff
@@ -17,6 +17,7 @@
   </PropertyGroup>  
   <ItemGroup>
     <PackageReference Include="Azure.AI.TextAnalytics" Version="5.1.0" />
+    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.5.0" />
     <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus" Version="4.3.0" />
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/FetchTranscription/FetchTranscription.csproj:21]: The Microsoft.Azure.WebJobs.Extensions.ServiceBus dependency needs to be updated to 5.1.0.

File: samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs (hunk 4 above)
Comment (v-yilinhu) [reply samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs:151]: @JoshLove-msft, There are ServiceBusMessage class in both Connector and Azure.Messaging.ServiceBus.

File: samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs (hunk 11 above)
Comment (v-yilinhu) [reply samples/ingestion/ingestion-client/StartTranscriptionByTimer/StartTranscriptionHelper.cs:275]: @JoshLove-msft, There are ServiceBusMessage class in both Connector and Azure.Messaging.ServiceBus.

**Comment by v-yilinhu** [comment]
@JoshLove-msft I have updated by your comments, please re-review this PR.

**Comment by v-xuto** [comment]
@JoshLove-msft We have updated by your comments, please re-review this PR.

**Comment by v-xuto** [comment]
@JoshLove-msft We have updated by your comments, please re-review this PR.

File: samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs [hunk 14]
```diff
@@ -8,26 +8,26 @@ namespace Connector
...
     using Microsoft.Extensions.Logging;
 
     public static class ServiceBusUtilities
     {
-        public static async Task SendServiceBusMessageAsync(QueueClient queueClient, string messageContent, ILogger log, TimeSpan delay)
+        public static async Task SendServiceBusMessageAsync(ServiceBusSender queueSender, string messageContent, ILogger log, TimeSpan delay)
         {
-            var message = new Message(Encoding.UTF8.GetBytes(messageContent));
-            await SendServiceBusMessageAsync(queueClient, message, log, delay).ConfigureAwait(false);
+            var message = new Azure.Messaging.ServiceBus.ServiceBusMessage(messageContent);
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs:18]:
```suggestion
            var message = new ServiceBusMessage(messageContent);
```

File: samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs [hunk 15]
```diff
@@ -155,8 +158,8 @@ private static async Task ProcessFailedTranscriptionAsync(string transcriptionLo
                         RetryCount = audio.RetryCount + 1
                     };
 
-                    var audioFileMessage = new Message(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(sbMessage)));
-                    await ServiceBusUtilities.SendServiceBusMessageAsync(StartQueueClientInstance, audioFileMessage, log, TimeSpan.FromMinutes(1)).ConfigureAwait(false);
+                    var audioFileMessage = new Azure.Messaging.ServiceBus.ServiceBusMessage(JsonConvert.SerializeObject(sbMessage));
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs:161]:
```suggestion
                    var audioFileMessage = new ServiceBusMessage(JsonConvert.SerializeObject(sbMessage));
```

File: samples/ingestion/ingestion-client/StartTranscriptionByServiceBus/StartTranscriptionByServiceBus.csproj [hunk 16]
```diff
@@ -16,7 +16,8 @@
     <OutputPath>bin\Release\</OutputPath>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.ServiceBus" Version="4.3.0" />
+    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.5.0" />
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/StartTranscriptionByServiceBus/StartTranscriptionByServiceBus.csproj:19]: Remove this and just use the extensions reference.

File: samples/ingestion/ingestion-client/Connector/Connector.csproj [hunk 17]
```diff
@@ -18,8 +18,9 @@
...
```
Comment (JoshLove-msft) [inline samples/ingestion/ingestion-client/Connector/Connector.csproj:21]: Remove this and just use the extensions reference.

File: samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs (hunk 15 above)
Comment (v-yilinhu) [reply samples/ingestion/ingestion-client/FetchTranscription/TranscriptionProcessor.cs:161]: @JoshLove-msft, There are ServiceBusMessage class in both Connector and Azure.Messaging.ServiceBus. So I need to distinguish them.

File: samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs (hunk 14 above)
Comment (v-yilinhu) [reply samples/ingestion/ingestion-client/Connector/ServiceBusUtilities.cs:18]: @JoshLove-msft, There are ServiceBusMessage class in both Connector and Azure.Messaging.ServiceBus. So I need to distinguish them.

**Comment by v-xuto** [comment]
@JoshLove-msft We have updated by your comments, please re-review this PR.

**Comment by v-xuto** [comment]
@JoshLove-msft We have updated by your comments, please re-review this PR. Thanks a lot.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by v-xuto** [comment]
@jongio  Please help merge this PR. Thanks a lot.

**Comment by jongio** [comment]
Please rebase @v-xuto

**Comment by v-xuto** [comment]
@jongio We have rebased the code. If there is no problem, please help merge the PR. Thanks a lot.

**Review by jongio — APPROVED** [review]

**Pull request merged**